great_expectations/render/renderer/v3/suite_profile_notebook_renderer.py

```diff
@@ -33,64 +34,52 @@ def __init__(
             batch_request=batch_request
         )
 
-        self.batch_request = batch_request
-
-        self.validator = context.get_validator(
+        self._batch_request = batch_request
+        self._validator = context.get_validator(
             batch_request=BatchRequest(**batch_request),
             expectation_suite_name=expectation_suite_name,
         )
 
-        self.expectation_suite_name = self.validator.expectation_suite_name
+        self._profiler_name = profiler_name
+        self._expectation_suite_name = self._validator.expectation_suite_name
 
     # noinspection PyMethodOverriding
-    def add_header(self) -> None:
-        self.add_markdown_cell(
-            markdown=f"""# Initialize a new Expectation Suite by profiling a batch of your data.
-This process helps you avoid writing lots of boilerplate when authoring suites by allowing you to select columns and other factors that you care about and letting a profiler write some candidate expectations for you to adjust.
+    def render(self) -> nbformat.NotebookNode:
+        self._notebook = nbformat.v4.new_notebook()
 
-**Expectation Suite Name**: `{self.expectation_suite_name}`
-"""
-        )
-        self.add_code_cell(
-            code=f"""\
-import datetime
+        self.add_header()
 
-import pandas as pd
+        # As v0.14.6, the notebook generated by `--profile` supports both the legacy UserConfigurableProfiler and the productionalized RuleBasedProfiler.
+        if self._profiler_name:
+            self._add_rule_based_profiler_cells()
+        else:
+            self._add_user_configurable_profiler_cells()
```

Comment:
While much of the core logic is the same, there are some key differences with imports, setup, and the actual profiler call.

This doesn't feel the cleanest but is necessary to leverage this dual profiler support. If we move away from the legacy approach, this conditional goes away and the implementation can become a bit cleaner.